src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs

```diff
@@ -350,6 +350,32 @@ public static IntPtr AllocateTypeAssociatedMemory(Type type, int size)
                 }
             }
         }
+
+#pragma warning disable 0414
+        internal unsafe ref struct GCFrameRegistration
+        {
+            private nuint m_reserved1;
+            private byte* m_pObjRefs;
+            private uint m_numObjRefs;
+            private nuint m_reserved2;
+            private int m_MaybeInterior;
```

Comment:
We can avoid holes by placing `m_numObjRefs` and `m_MaybeInterior` next to each other. (It requires doing the same change in unmanaged `GCFrame`.